scip.proto

```diff
@@ -383,10 +383,31 @@ message Occurrence {
   // type with `start` and `end` fields of type `Position`, mirroring LSP.
   // Benchmarks revealed that this encoding was inefficient and that we could
   // reduce the total payload size of an index by 50% by using `repeated int32`
-  // instead.  The `repeated int32` encoding is admittedly more embarrassing to
+  // instead. The `repeated int32` encoding is admittedly more embarrassing to
   // work with in some programming languages but we hope the performance
   // improvements make up for it.
   repeated int32 range = 1;
+  // (optional) The source position of the nearest non-trivial enclosing AST node.
+  // If supplied, the value must mirror the constraints detailed in the `range` field.
+  // For occurrences referencing the identifier within a class or function declaration, 
+  // the  obvious interpretation of this field is to encode the bounds of the entire
+  // declaration, e.g. in the following the enclosing range captures the `func` keyword
+  // to the function declaration's closing brace.
+  // ```
+  // const n = 3
+  //
+  // func Parse(input string) string {
+  //     return input[n:]             
+  // }                                
+  // ```
+  // This field can also be used, for further example, to indicate the relevant parent 
+  // expression a variable referencing an invokable, e.g.:
+  // ```
+  // let x = a.b().f().g()
+  //               ^ range
+  //         ^^^^^^^ enclosing_range
```

Comment:
My expectation is that the enclosing range would include the function arguments as well
```suggestion
  //         ^^^^^^^^^ enclosing_range
```